src/components/TodoApp/TodoApp.tsx

```diff
@@ -0,0 +1,252 @@
+import React, {
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+import { useParams } from 'react-router-dom';
+import { Status } from '../../types/Status';
+import { Todo } from '../../types/Todo';
+import { useLocalStorage } from '../../utils/useLocalStorage';
+import { TodoList } from '../TodoList';
+import { TodosFilter } from '../TodosFilter';
+
+export const TodoApp = () => {
+  const [newTodoTitle, setNewTodoTitle] = useState<string>('');
+  const [todos, setTodos] = useLocalStorage<Todo[]>('todos', []);
+
+  const [copyTodos, setCopyTodos] = useLocalStorage<Todo[]>('todos', []);
+
+  const [changedTitleTodoId, setChangedTitleTodoId]
+    = useState<number | null>(null);
+
+  const { status } = useParams();
+
+  const activeTodosLength = useMemo(() => (
+    copyTodos.filter(el => !el.completed).length
+  ), [copyTodos]);
+
+  const [activeToggleAll, setActiveToggleAll]
+    = useState(false);
+  const [newEditedTitle, setNewEditedTitle] = useState('');
+
+  useEffect(() => {
+    setActiveToggleAll(activeTodosLength === 0);
+  }, [activeTodosLength]);
+
+  const completedTodos
+    = useMemo(() => (copyTodos.filter(todo => todo.completed)),
+      [copyTodos]);
+
+  const createNewTodo = (title: string): Todo => ({
+    id: +new Date(),
+    title,
+    completed: false,
+  });
+
+  const handleInputChange
+  = (event: React.ChangeEvent<HTMLInputElement>, param: string) => {
+    const { value } = event.target;
```

Comment:
```suggestion
  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>, param: string) => {
    const { value } = event.target;
```

Reply:
theare are more than 80 symbols here, so i moved the line